chef_master/source/environments.rst

```diff
@@ -53,16 +53,11 @@ There are two types of attributes that can be used with environments:
 
        .. end_tag
 
-Attribute Persistence
+Attribute Precedence
 -----------------------------------------------------
-.. tag node_attribute_persistence
-
-All attributes except for normal attributes are reset at the beginning of a Chef Infra Client run. Chef Infra Client rebuilds these attributes using automatic attributes collected by Ohai at the beginning of each Chef Infra Client run, and then uses default and override attributes that are specified in cookbooks, roles, environments, and Policyfiles. All attributes are then merged and applied to the node according to attribute precedence. The attributes that were applied to the node are saved to the Chef Infra Server as part of the node object at the conclusion of each Chef Infra Client run.
 
-.. end_tag
+Environments are one of several locations where attributes can be applied in Chef Infra. It's important to understand how the precedence level works in order to understand, which attributes will be applied when Chef Infra Client runs.
```

Comment:
Environments are one of several different places [FILES?] for defining attributes in Chef Infra. Attributes are defined in cookbook attribute files, recipes, environments, roles, and also in Ohai.  Attribute _precedence_ explains why an attribute defined in one file wins out over another during a Chef Infra Client run and gets applied to a node.

Reply:
I think we should reverse the order of the attribute precedence list, so #1 is the Ohai automatic attribute.